src/main/scala/eu/proteus/solma/lasso/LassoParameterServer.scala

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2017 The Proteus Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.proteus.solma.lasso
+
+import breeze.linalg._
+import hu.sztaki.ilab.ps.entities.{PSToWorker, Pull, Push, WorkerToPS}
+import hu.sztaki.ilab.ps.server.{RangePSLogicWithClose, SimplePSLogicWithClose}
+import hu.sztaki.ilab.ps.{FlinkParameterServer, ParameterServerClient, WorkerLogic}
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.streaming.api.scala._
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+import eu.proteus.solma.lasso.algorithm.LassoAlgorithm
+import eu.proteus.solma.lasso.Lasso.{LassoModel, LassoParam, OptionLabeledVector}
+import eu.proteus.solma.lasso.algorithm.LassoParameterInitializer._
+
+class LassoParameterServer
+
+
+object LassoParameterServer {
+
+  /**
+    * Applies online Lasso algorithm for a [[org.apache.flink.streaming.api.scala.DataStream]] of vectors.
+    * For vectors with labels, it updates the Lasso model,
+    * for vectors without label predicts its labels based on the model.
+    *
+    * Note that the order could be mixed, i.e. it's possible to predict based on some model parameters updated, while
+    * others not.
+    *
+    * @param inputSource
+    * [[org.apache.flink.streaming.api.scala.DataStream]] of labelled and unlabelled vector. The label is marked with an
+    * [[Option]].
+    * @param workerParallelism
+    * Number of worker instances for Parameter Server.
+    * @param psParallelism
+    * Number of Parameter Server instances.
+    * @param lassoMethod
+    * Method for Lasso training.
+    * @param pullLimit
+    * Limit of unanswered pulls at a worker instance.
+    * @param iterationWaitTime
+    * Time to wait for new messages at worker. If set to 0, the job will run infinitely.
+    * PS is implemented with a Flink iteration, and Flink does not know when the iteration finishes,
+    * so this is how the job can finish.
+    * @return
+    * Stream of Parameter Server model updates and predicted values.
+    */
+  def transformLasso(model: Option[DataStream[(Int, LassoModel)]] = None)
+                    (inputSource: DataStream[OptionLabeledVector],
```

Comment:
Solma features [StreamEvent](https://github.com/proteus-h2020/proteus-solma/blob/master/src/main/scala/eu/proteus/solma/events/StreamEvent.scala) as main data object as we wanted to support proteus scenario (i.e., a stream event contains a sparse vector). Do you think we could eventually integrate StreamEvent and LabeledVector?